### DISCUSSION TITLE
MultiApps - multi core questions

### DISCUSSION THREAD

**Discussion by MatiasAllay** [idaholab/moose#18098] [Q&A Modules: General] [answered]
Hi MOOSE,
I'm currently running a simulation using `Multiapps`. The mode is batch-reset and as a sampler I'm using a `CSVSampler` that has only two columns but many rows. I'm also using `MultiAppCommandLineControl` to change materials parameters as it was suggested here https://github.com/idaholab/moose/discussions/18046. My problem now is that even though I'm using multiple cores when I call the simulation, for each simulation MOOSE is only using one core for each "multiapp" (and running one "multiapp" at time). Any advice in how can I force MOOSE to use all the available cores?
Cheers,
Matias

**Comment by jiangwen84** [comment]
By default, each multiapp  uses 1 processor. You can set `min_procs_per_app` in your `MultiApp` block  to use multiple processors for each multiapp.

**Comment by GiudGiud** [comment]
Hello

Have you tried using the min_procs_per_app and min_procs_per_row parameters?
See documentation there https://mooseframework.inl.gov/source/multiapps/SamplerFullSolveMultiApp.html

Guillaume

**Comment by MatiasAllay** [comment]
Hi MOOSE,
Thanks for your help! I added  `min_procs_per_app` in my MultiApps block and also `min_procs_per_row` in my Samplers block as is suggested in the documentation (https://mooseframework.inl.gov/source/multiapps/SamplerFullSolveMultiApp.html) and I set it with the number of processors I want to use, but nothing happens (simulation still runs with one processor although MOOSE "blocks" all the available processors). 
Any idea of what else could I try?
Thanks!!
Matias

**Reply by aeslaughter** [reply]
@zachmprince Might have some insight into what is happening, if I recall he implemented this capability.

**Reply by MatiasAllay** [reply]
Hi @aeslaughter how could we proceed? Shall we move this issue somewhere else? Thanks.

**Comment by fdkong** [comment]
>Thanks for your help! I added min_procs_per_app in my MultiApps block and also min_procs_per_row in my Samplers block as is suggested in the documentation (https://mooseframework.inl.gov/source/multiapps/SamplerFullSolveMultiApp.html) and I set it with the number of processors I want to use, but nothing happens (simulation still runs with one processor although MOOSE "blocks" all the available processors).

Could you post the multiapp block of your input file?  I went through `SamplerFullSolveMultiApp`, but  I did not see anything special.

**Comment by zachmprince** [comment, marked as answer]
Apologies, I just noticed this discussion. I created an input using a SamplerCSV. You can find the main and sub inputs [here](https://github.com/zachmprince/moose/commit/597e69177fb82592f1cf9410e6478ba2addb001b). It seemed to work fine with `min_procs_per_app` and using `mpiexec -n 4`. It would be helpful to see your input though.

**Comment by MatiasAllay** [comment]
Hi all,
Thanks for your kind reply. Thanks @zachmprince for the example. I missed the `StochasticTools` block in my subApp: 
[StochasticTools]
[]
Now my simulation seems to run fine!
Thanks
Matias

**Reply by zachmprince** [reply]
Be careful with that syntax. If you use that, the problem won't actually solve. I put that in my sub app input because I'm just testing the partitioning and not solving a problem.

**Reply by MatiasAllay** [reply]
You are right... I got excited that my simulation was running fast with  `[StochasticTools]` in the subApp that I didn't noticed that nothing was happening :(. But today I tried running your example and I got an updated warning, so I updated and now seems to work fine!
Thanks again!
Matias
